assignments/lambda-classes.js

```diff
@@ -1 +1,140 @@
 // CODE here for your Lambda Classes
+
+// Base class
+class Person {
+    constructor(attributes) {
+        this.name = attributes.name;
+        this.age = attributes.age;
+        this.location = attributes.location;
+        this.gender = attributes.gender;
+    }
+    speak() {
+         return `Hello my name is ${this.name}, I am from ${this.location}`;
+    }
+}
+
+// Instructor class
+class Instructor extends Person {
+    constructor(instAttributes) {
+        super(instAttributes);
+        this.specialty = instAttributes.specialty;
+        this.favLanguage = instAttributes.favLanguage;
+        this.catchPhrase = instAttributes.catchPhrase;
+    }
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+    grade(student, subject) {
+        return `${student.name} receives a perfect score on ${subject}`;
+    }
+    addPoints(student) {
+        return `${student.name} gets points of ${Math.floor(Math.random() * 6) + student.score}`;
+    }
+    subtractPoints(student) {
+        return `${student.name} gets points of ${student.score - Math.floor(Math.random() * 6)}`;
+    }
+}
+
+// Student class
+class Student extends Person {
+    constructor(stuAttributes) {
+        super(stuAttributes);
+        this.previousBackground = stuAttributes.previousBackground;
+        this.className = stuAttributes.className;
+        this.favSubjects = stuAttributes.favSubjects;
+        this.score = stuAttributes.score;
+    }
+    listsSubjects() {
+        return `Favorite subjects are: ${this.favSubjects.join(' --')}`;
```

Comment:
For this one he did actually want a for loop. However, you did a great job. perfect use of `join`